tests/test_special.py

```diff
@@ -54,116 +61,97 @@ def tearDownClass(cls):
+
+    def test_show_function_definition_command(self, client):
+        for rows, col, _, _, _ in client.execute_query('\\sf {0}'.format(self.function)):
+            assert len(rows) == 1
+            assert len(col) == 1
+
+    def test_describe_object_command(self, client):
+        result_set_count = 0
+        for _ in client.execute_query('\\d {0}'.format(self.function)):
+            result_set_count += 1
+
+        assert result_set_count == 2
+
+    @staticmethod
+    def test_named_queries_commands(client):
+        # Save named queries
+        list(client.execute_query('\\sn test123 select 1'))
+        list(client.execute_query('\\sn test234 select 2'))
+
+        # List named queries
+        for rows, col, _, _, _ in client.execute_query('\\n'):
+            assert len(rows) >= 2
+            num_queries = len(rows)
+
+        # Execute named query created above
+        for rows, col, _, _, _ in client.execute_query('\\n test123'):
+            assert len(rows) == 1
+            assert len(col) == 1
+
+        # Delete a named query that was created
+        list(client.execute_query('\\dn test123'))
+
+        # Number of named queries should have reduced by 1
+        for rows, col, _, _, _ in client.execute_query('\\n'):
+            assert num_queries-1 == len(rows)
+
+        # Clean up the second named query created
+        list(client.execute_query('\\dn test234'))
+
+    @staticmethod
+    def test_add_new_special_command():
         @special_command('\\empty', '\\empty[+]', 'returns an empty list', arg_type=NO_QUERY)
         def empty_list_special_command():
             # pylint: disable=unused-variable
             return []
 
         ret = execute(None, '\\empty')
-        self.assertTrue(len(ret) == 0)
+        assert len(ret) == 0
 
-    def command(self, command, pattern, min_rows_expected, rows_expected_pattern_query,
+    @staticmethod
+    def command(client, command, pattern, min_rows_expected, rows_expected_pattern_query,
                 cols_expected, cols_expected_verbose):
         # pylint: disable=too-many-arguments
-        try:
-            client = create_mssql_cli_client()
-
-            for rows, col, _, _, _ in client.execute_query(command):
-                self.assertTrue(len(rows) >= min_rows_expected)
-                self.assertTrue(len(col) == cols_expected)
-
-            # execute with pattern and verbose
-            command = command + "+ " + pattern
-            for rows, col, _, _, _ in client.execute_query(command):
-                self.assertTrue(len(rows) == rows_expected_pattern_query)
-                self.assertTrue(len(col) == cols_expected_verbose)
-        finally:
-            shutdown(client)
 
+        for rows, col, _, _, _ in client.execute_query(command):
+            assert len(rows) >= min_rows_expected
+            assert len(col) == cols_expected
 
-if __name__ == u'__main__':
```

Comment:
These main's were useful to run files :(

Reply:
I can put this back!